src/pkg/cluster/state.go

```diff
@@ -277,10 +283,14 @@ func (c *Cluster) MergeZarfState(oldState *types.ZarfState, initOptions types.Za
 
 		// Set the new passwords if they should be autogenerated
 		if newState.GitServer.PushPassword == oldState.GitServer.PushPassword && oldState.GitServer.InternalServer {
-			newState.GitServer.PushPassword = utils.RandomString(config.ZarfGeneratedPasswordLen)
+			if newState.GitServer.PushPassword, err = helpers.RandomString(config.ZarfGeneratedPasswordLen); err != nil {
+				message.Fatal(err, lang.ErrUnableToGenerateRandomSecret)
+			}
 		}
 		if newState.GitServer.PullPassword == oldState.GitServer.PullPassword && oldState.GitServer.InternalServer {
-			newState.GitServer.PullPassword = utils.RandomString(config.ZarfGeneratedPasswordLen)
+			if newState.GitServer.PullPassword, err = helpers.RandomString(config.ZarfGeneratedPasswordLen); err != nil {
+				message.Fatal(err, lang.ErrUnableToGenerateRandomSecret)
+			}
 		}
```

Comment:
Yeah I set it to fatal as I was moving the `message.fatal` out of utils.randomstring. I like your explanation in thinking about it as a library and handing the fatal in the cmd package. Will change it to that

Reply:
Made that change. One thing I wasn't sure about was the following function calls in the same file. I could see either copying the initOptions.GitServer to state.GitServer and sending in a pointer or just sending in and receiving a value as I did here. Functionally I believe it's exactly the same not sure if we prefer the style of one over the other. @Racer159 @Noxsios 

```
		if state.GitServer, err = c.fillInEmptyGitServerValues(initOptions.GitServer); err != nil {
			return err
		}
		if state.RegistryInfo, err = c.fillInEmptyContainerRegistryValues(initOptions.RegistryInfo); err != nil {
			return err
		}
```

Reply:
After talking with Wayne we decided both styles were reasonable but in this case since we preferred not sending it as a pointer since it's only going to one function and it'll keep it less side-affecty